metalgpy/_sample.py

```diff
@@ -36,12 +38,104 @@ def list_type(list_exp):
             return LIST_KCATEGORICAL
 
 class BaseSampler:
-    def __init__(self, exp, distributions: dict=None, rng:np.random.RandomState=None):
+    """Defines a Base Sampler
+    
+    Args:
+        exp (Expression): a metalgpy expression.
+        distributions (dict, optional): preferred distributions for drawing samples. Defaults to None.
+        rng (int, optional): a random seed to generate same stream of values.Defaults to None.
+    """
+    def __init__(self,
+                 exp,
+                 distributions: dict=None,
+                 rng:np.random.RandomState=None):
         self.exp = exp
-        self.distributions = distributions
+        self.distributions = self._map_dist(distributions)
         self.rng = rng
+        self._cache = {}
+
+    def sample(self, size=None, to_nd_array=True):
+        # add a list to store samples
+        samples = []
+
+        # init a random state is rng
+        if self.rng is None:
+            self.rng = np.random.RandomState()
 
+        # check if the size is None
+        if size is None:
+            size = 1
 
+        # gather the list of choices
+        choices = self.exp.choices()
+
+        # get the required number of samples
+        for _ in range(size):
+            choice = self._sample(choices)
+
+            # store the sample
+            samples.append(choice)
+
+        # add a try catch block to capture returning elements to an nd-array
+        try:
+            if to_nd_array:
+                # convert the samples to an nd_array
+                samples = np.array(samples).reshape(len(samples), -1)
+        except ValueError:
+            # warn about invalid array structure
+            print("Cannot convert the sample to a numpy array")
+
+        # return the list in a numpy array
+        return samples
+
+    def _sample(self, choices):
+        # init a list to store samples
+        s = []
+
+        for var_id, var_exp in choices.items():
+            # check if the expression is Float
+            if isinstance(var_exp, Float):
+                # add samples
+                s.append(self.distributions[var_id].rvs(loc=var_exp._low, \
```

Comment:
this mechanic will break if a user decide to pass an other distribution, for example normal distribution